maven-core/src/main/java/org/apache/maven/project/DefaultProjectBuilder.java

```diff
@@ -1043,6 +1115,120 @@ private String findProfilesXml( ModelBuildingResult result, Map<File, Boolean> p
 
         return null;
     }
+    
+    private Collection<FileTransformer> getTransformersForArtifact( final org.eclipse.aether.artifact.Artifact artifact,
+                                                                    TransformerContext context )
+    {
+        Collection<FileTransformer> transformers = new ArrayList<>();
+        if ( "pom".equals( artifact.getExtension() ) )
+        {
+            final SAXTransformerFactory transformerFactory =
+                (SAXTransformerFactory) Factories.newTransformerFactory();
+            
+            transformers.add( new FileTransformer()
+            {
+                @Override
+                public InputStream transformData( File file )
+                    throws IOException, TransformException
+                {
+                    final PipedOutputStream pipedOutputStream  = new PipedOutputStream();
+                    final PipedInputStream pipedInputStream  = new PipedInputStream( pipedOutputStream );
+                    
+                    final TransformerHandler transformerHandler =
+                        getTransformerHandler( transformerFactory, file );
+                    
+                    BuildPomXMLFilterFactory buildPomXmlFactory = new DefaultBuildPomXMLFilterFactory( context );
+                    
+                    final SAXSource transformSource;
+                    try
+                    {
+                        AbstractSAXFilter filter =
+                            new ConsumerPomXMLFilterFactory( buildPomXmlFactory ).get( file.toPath() );
+                        filter.setLexicalHandler( transformerHandler );
+                        
+                        transformSource =
+                            new SAXSource( filter, new InputSource( new FileInputStream( file ) ) );
+                    }
+                    catch ( SAXException | ParserConfigurationException | TransformerConfigurationException e )
+                    {   
+                        throw new TransformException( "Failed to create a consumerPomXMLFilter", e );
+                    }
+                    
+                    transformerHandler.setResult( new StreamResult( pipedOutputStream ) );
+                    
+                    SAXResult transformResult = new SAXResult( transformerHandler );
+                    
+                    ExecutorService executorService = Executors.newSingleThreadExecutor();
+                    executorService.execute( () -> 
```

Comment:
What is the reason to run this async? Who is receiving this `pipedInputStream`?

Reply:
Good catch, I should use the thread implementation here.